appserver/web/web-ha/src/main/java/org/glassfish/web/ha/authenticator/HASingleSignOnEntry.java

```diff
@@ -159,50 +159,23 @@ public long incrementAndGetVersion() {
     /** convert a principal into byte array */
     private byte[] convertToByteArray(Principal obj) {
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        BufferedOutputStream bos = null;
-        ObjectOutputStream oos = null;
-        try {
-            bos = new BufferedOutputStream(baos);
-            oos = ioUtils.createObjectOutputStream(bos, true);
+        try (ObjectOutputStream oos = ioUtils.createObjectOutputStream(new BufferedOutputStream(baos), true)) {
             oos.writeObject(obj);
             oos.flush();
             return baos.toByteArray();
         } catch(Exception ex) {
             throw new IllegalStateException("Could not convert principal to byte array", ex);
-        } finally {
-          closeSilently(baos);
```

Comment:
A `ByteArrayOutputStream` does not have to be closed. Its close is a NOOP anyway. Therefore there does not need to be another level of try for the `baos`.

Reply:
It cannot be within the block as it is used both in the resource construction expression and within the try block. If we want it covered it would need its own level of try-with-resouce.

Reply:
try can accept multiple assignment statements.

Reply:
Learned about multi-variable try-with resources ;) Thanks @pdudits . Whet ahead and used it. As discussed I also removed the buffer wrapping as well that in this case should just cause overhead for no benefit since a `ByteArrayOutputStream` already is a "buffer".